bot/code_review_bot/revisions.py

```diff
@@ -353,19 +353,31 @@ def from_phabricator_trigger(
                 diff_phid = new
                 break
 
-        # Check a diff is found
+        # Check a diff is found in transactions or use last diff available
         if diff_phid is None:
```

Comment:
How can this happen? Is there a risk we trigger a task, in the meantime the developer uploads a new patch, we trigger another one, and so in the end we have two tasks running for the same diff?

Reply:
I do not know why Phabricator did not send the Diff creation transaction in the list of transactions, but it was really missing.

You are right, there is currently no mutex-like mechanism to prevent running multiple times. We could probably achieve that by using Taskcluster routes (setting a route in a task, and checking it does not exist at task start).